drivers/note/note_driver.c

```diff
@@ -1393,10 +1393,12 @@ void sched_note_irqhandler(int irq, FAR void *handler, bool enter)
 void sched_note_wdog(uint8_t event, FAR void *handler, FAR const void *arg)
 {
   FAR struct note_driver_s **driver;
+  irqstate_t flags;
   struct note_wdog_s note;
   bool formatted = false;
   FAR struct tcb_s *tcb = this_task();
 
+  flags = enter_critical_section();
```

Comment:
Why is wdog special?

Reply:
sched_note_wdog should be protected by critical_section, but since we have removed the critical_section in wdog, we need to protect it separately